### PR TITLE
Upgrade intervention service docker java 17

### PR DESCRIPTION
**Pull request by kth13**
## What does this pull request do?

- upgrades the intervention service to JDK17. Updates the docker image to use openjdk-17

## What is the intent behind these changes?

It is a technical debt to upgrade intervention service to latest jdk version.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,17 +21,17 @@ RUN ./gradlew assemble
 
 
 # ---
-FROM adoptopenjdk/openjdk11:alpine-jre
+FROM openjdk:17-slim
```
[the remark this answers is not in the document]
Comment (kth13) [reply Dockerfile:24]: I am not able to find the  **adoptopenjdk/openjdk17:alpine-jre** from the docker hub.What I found was adoptopenjdk was deprecated. I was checking other repositories and they were using openjdk:17-slim. That is the reason why I choose this version

[the remark this answers is not in the document]
Comment (tomsmyers) [reply Dockerfile:24]: awesome - fine with me. would still be good to see how it affects the size of the final build.

**Review by tomsmyers — APPROVED** [review]

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (kth13) [reply Dockerfile]: The application sizes built with the following base images
eclipse-temurin:17.0.2_8-jre-focal ==> 289MB
alpine:3.15 ==> 349MB

File: Dockerfile [hunk 2]
```diff
@@ -21,17 +21,18 @@ RUN ./gradlew assemble
 
 
 # ---
-FROM adoptopenjdk/openjdk11:alpine-jre
+FROM alpine:3.15 AS final
 LABEL maintainer="HMPPS Digital Studio <info@digital.justice.gov.uk>"
 
-# force a rebuild of `apk upgrade` below by invalidating the BUILD_NUMBER env variable on every commit
+# force a rebuild of `apt-get update` below by invalidating the BUILD_NUMBER env variable on every commit
```
Comment (sldblog) [inline Dockerfile:27]: 🙋‍♂️ This change is no longer needed (apk stayed below)

**Review by sldblog — CHANGES_REQUESTED** [review]
CircleCI config also has some Java 11 references, eg.

https://github.com/ministryofjustice/hmpps-interventions-service/blob/f70506b5698ec56724021d0686126ae80298daca/.circleci/config.yml#L17

~~Unsure how to make sure that all jobs use the Java 17 executor~~

It looks like the "tag" can be given for the `hmpps/java` executor: https://github.com/ministryofjustice/hmpps-circleci-orb/blob/be1dbcd551d4e8771f026ced818bd38baad035aa/src/executors/java.yml#L3

Could those be updated, please?

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (sldblog) [reply Dockerfile]: Do we save with the `openjdk17-jre-headless` package? That one doesn't install the sound and graphical libraries

**Comment by sldblog** [comment]
Yeah, I think we will have to do something like this

https://github.com/ministryofjustice/hmpps-assess-risks-and-needs/blob/589df543dabc3d6b6c625b099e85d2b9d2be0b66/.circleci/config.yml#L13-L15

By default, the non-dockerised Java builds would run on an executor which has Java 11 so we could have a pass where Java 17 would fail

**Comment by sldblog** [comment]
Please ignore the above -- I didn't see db39bb69b927423d343a1e8f2117c7fca82f7624 commit before I made the above comment about 17.0

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (kth13) [reply Dockerfile]: alpine:3.15 ==> openjdk17-jre ==> 349MB
alpine:3.15 ==> openjdk17-jre-headless ==> 349MB

In terms of size, both are same

[the remark this answers is not in the document]
Comment (sldblog) [reply Dockerfile]: 👍 that's weird, but ok

can we still switch to it, please? fewer dependencies mean fewer security CVEs

[the remark this answers is not in the document]
Comment (kth13) [reply Dockerfile]: @sldblog , sure I will change to it

**Review by sldblog — APPROVED** [review]
👍 LGTM

To note, this should still produce JVM11-compatible bytecode as-is

**Pull request merged**